pkg/kubectl/cmd/rollout/rollout_history.go

```diff
@@ -49,6 +49,9 @@ var (
 func NewCmdRolloutHistory(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	options := &HistoryOptions{}
 
+	validArgs := []string{"deployment"}
```

Comment:
> Just noticed it, thanks. Are we going to end up with a factory method that needs to separate between commands?

Maybe.  I see what's been done here as strictly better than the current state of affairs.  I'd let this go in and @fabianofranz and team can fix it up in the 1.5 timeframe.